Add array support to `where` filter

timwis opened the pull request:
In addition to `where` supporting basic string matches (ie. `category == "foo"`) this allows the `where` filter to work on array values, ie. `category contains "foo"`. So whether the page has
```yaml
---
category: Foo
---
```
or
```yaml
---
category:
  - Foo
---
```
The `where` filter will still work (`{{ site.pages | where:"category", "foo" }}`)

Note that this should not have an effect on any existing (string-match) use of `where` (non-breaking change)

Resolves #4385 (feature request)

parkr commented:
This changes the behaviour of this filter in possibly nefarious ways because it'd be grabbing things that previously it didn't. @benbalter, what say you? If we can't come up with an example where it'd be a problem, then I'm :+1:.

benbalter commented:
I can't think of an example where this would break existing behavior (also :wave: @timwis).

timwis commented:
Hey @benbalter! Thanks for the review guys.

timwis commented:
Okay, figured out why the tests were failing and rewrote the PR to account for all property types. I also added 2 tests. I'm new to ruby so please let me know if I've made any convention mistakes or anything.

parkr commented:
What happens if you have `category: bear` and you search for `where: "category", "ear"`? From my reading of the code, you'd get back that post with `category: bear` even though you weren't really interested in that.

timwis commented:
Hey @parkr, perhaps I'm misunderstanding, but I don't think that's the case. I've added a test for it to demonstrate. Let me know if I'm off.

parkr commented:
This LGTM. Thanks!

@jekyllbot: merge +minor

parkr commented:
@jekyllbot: merge +minor

parkr commented:
@jekyllbot: merge +minor

parkr commented:
@jekyllbot: merge +minor
